### ISSUE TITLE
Linking fails with OpenSSL 1.0.2

### ISSUE THREAD

**Issue by estaban** [sfackler/rust-openssl#1588]
Wnen compiling a Rust application that depends on openssl 0.10.38 using against OpenSSL 1.0.2 binaries, linking fails with the following errors:

```
undefined reference to `SSL_CTX_set_tlsext_use_srtp'
undefined reference to `SSL_set_tlsext_use_srtp'
undefined reference to `SSL_get_srtp_profiles'
undefined reference to `SSL_get_selected_srtp_profile'
```

Adding `#[cfg(ossl111)]` above these functions in https://github.com/sfackler/rust-openssl/blob/master/openssl/src/ssl/mod.rs solves the compilation problem.

**Comment by sfackler** [comment]
They are definitely present in 1.0.2: https://github.com/openssl/openssl/blob/OpenSSL_1_0_2u/ssl/srtp.h#L135.

It looks like we just need to add detection of `OPENSSL_NO_SRTP`.

**Comment by jonhoo** [comment]
This hit me today since I'm using a hardened version of OpenSSL that doesn't have these functions (well, `SSL_get_selected_srtp_profile` in particular at least). @sfackler do you have a good sense of where the changes should happen? I'm guessing something like make `build.rs` pick up on this variable and emit a `cfg`, and then use that `cfg` on any srtp-related functionality?

Also, should we re-title this issue to "Support OpenSSL with SRTP disabled" or something like that?

**Comment by sfackler** [comment]
Yep, the define needs to be added here: https://github.com/sfackler/rust-openssl/blob/master/openssl-sys/build/expando.c, and then the generated cfg needs to be use to flag off those APIs.
